### ISSUE TITLE
`useSearchParams` on iOS Safari makes the page scroll down

### ISSUE THREAD

**Issue by IonelLupu** [vercel/next.js#45610]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.2.0: Fri Nov 11 02:03:51 PST 2022; root:xnu-8792.61.2~4/RELEASE_ARM64_T6000
    Binaries:
      Node: 18.12.1
      npm: 8.19.2
      Yarn: 1.22.19
      pnpm: N/A
    Relevant packages:
      next: 13.1.7-canary.5
      eslint-config-next: 13.1.4
      react: 18.2.0
      react-dom: 18.2.0

### Link to the code that reproduces this issue

https://codesandbox.io/p/sandbox/confident-rumple-xz03rh?file=%2Fapp%2Fpage.tsx&selection=%5B%7B%22endColumn%22%3A10%2C%22endLineNumber%22%3A8%2C%22startColumn%22%3A10%2C%22startLineNumber%22%3A8%7D%5D

### To Reproduce

1. create a new next.js 13 project with default settings and the experimental app directory:
    `npx create-next-app@latest --experimental-app`
2. add a long piece of text in the `app/page.tsx` file. Also, add the `useSearchParams()` hook in the as well:
```
"use client";
import { useSearchParams } from "next/navigation";

export default function Home() {
  useSearchParams();
  return (
    <div>
      <p>scroll down to see a link and then click it!</p>

      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>
      <p>scroll down to see a link and then click it!</p>

      <br />
      <br />
      <br />
      <br />
      <br />
      <br />
      <br />

      <br />
      <br />
      <br />

      <br />
      <br />
      <br />
      <br />
      <br />
      <p>How did I get scrolled here???</p>
      <p>How did I get scrolled here???</p>
      <p>How did I get scrolled here???</p>

      <p>How did I get scrolled here???</p>
    </div>
  );
}

```

3. build the app: `yarn build` or `npm run build`
4. open the app on Safari on iOS: `localhost:3000`. I used the iPhone 14 Pro Max on the XCode simulator.
5. Watch how the scrollbar is scrolled a few hundred pixels:
![image](https://user-images.githubusercontent.com/4083652/216963920-721461ae-976b-4ada-bbab-346b86d24794.png)

### Describe the Bug

When using the `useSearchParams()` on a page and opening that page on iOS Safari on iPhone Pro Max after building the app, the page is scrolled down a few hundred pixels

### Expected Behavior

Expecting the page not to be scrolled.

Here is a video showing the before and after using `useSearchParams()`. The video is a bit glitchy but you will see the issue after the second `yarn start` and after I refresh the page (the scrollbar is in the middle of the page):

https://user-images.githubusercontent.com/4083652/216965520-dee47957-5705-4b94-9bcb-77c33a44d72a.mov

### Which browser are you using? (if relevant)

5.0 (iPhone; CPU iPhone OS 16_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.2 Mobile/15E148 Safari/604.1

### How are you deploying your application? (if relevant)

next start

**Comment by yaseenmustapha** [comment]
Can confirm I am encountering this bug as well

**Comment by frederikfink** [comment]
Any updates on this one? I'm also experiencing this bug

**Comment by marcoshernanz** [comment]
Thanks for the report. This reproduction calls `useSearchParams()` in a statically rendered page without a Suspense boundary, which caused the entire page to fall back to client rendering in Next 13. Current canary rejects this setup at build time and requires a [Suspense boundary](https://nextjs.org/docs/app/api-reference/functions/use-search-params#static-rendering), so the original reproduction is no longer applicable. If this still occurs on current canary with the supported setup, please open a new issue with an updated reproduction, iOS version, and whether it happens on first load or only after reload.
